src/js_native_api_v8.h

```diff
@@ -122,6 +122,37 @@ struct napi_env__ {
   void* instance_data = nullptr;
 };
 
+// This class is used to keep a napi_env live in a way that
+// is exception safe versus calling Ref/Unref directly
+class LiveEnv {
```

Comment:
Maybe we can call this `EnvRefHolder` or similar in a more explicit way?

Reply:
@gabrielschulhof, @addaleax, @KevinEady, thoughts on the name? Does EnvRefHolder sound good to you?

Reply:
I'm good with EnvRefHolder and see Anna was +1 as well. Will update.

Reply:
@legendecas, @addaleax updated if you want to take another look.